src/System.Memory/tests/Performance/Perf.ReadOnlySequence.Slice.cs

```diff
@@ -2,765 +2,338 @@
+        [InlineData(10_000, 10)]
+        public void PosLong(int bufSize, int posCount)
         {
-            var segment1 = new BufferSegment<byte>(new byte[bufSize / 10]);
-            BufferSegment<byte> segment2 = segment1;
-            for (int j = 0; j < 10; j++)
-                segment2 = segment2.Append(new byte[bufSize / 10]);
-            var buffer = new ReadOnlySequence<byte>(segment1, bufOffset, segment2, bufSize / 10 - bufOffset);
-            long offset = buffer.Length / 10;
+            ReadOnlySequence<T> buffer = Factory.CreateOfSize(bufSize);
 
+            List<(SequencePosition, long)> positions = CookPositionsPosLong(buffer, posCount);
             foreach (BenchmarkIteration iteration in Benchmark.Iterations)
             {
                 int localInt = 0;
                 using (iteration.StartMeasurement())
                 {
                     for (int i = 0; i < Benchmark.InnerIterationCount; i++)
                     {
-                        ReadOnlySequence<byte> temp = buffer.Slice(offset);
-                        localInt ^= temp.Start.GetInteger();
+                        foreach ((SequencePosition start, long length) in positions)
+                        {
+                            ReadOnlySequence<T> sliced = buffer.Slice(start, length);
+                            localInt ^= sliced.Start.GetInteger();
+                        }
                     }
                 }
                 _volatileInt = localInt;
             }
         }
 
         [Benchmark(InnerIterationCount = InnerCount)]
-        [InlineData(10_000, 100)]
-        private static void Byte_MultiSegment_LongLong(int bufSize, int bufOffset)
+        [InlineData(10_000, 10)]
+        public void PosPos(int bufSize, int posCount)
         {
-            var segment1 = new BufferSegment<byte>(new byte[bufSize / 10]);
-            BufferSegment<byte> segment2 = segment1;
-            for (int j = 0; j < 10; j++)
-                segment2 = segment2.Append(new byte[bufSize / 10]);
-            var buffer = new ReadOnlySequence<byte>(segment1, bufOffset, segment2, bufSize / 10 - bufOffset);
-            long offset = buffer.Length / 20;
-            long sliceLen = buffer.Length - 2 * offset;
+            ReadOnlySequence<T> buffer = Factory.CreateOfSize(bufSize);
 
+            List<(SequencePosition, SequencePosition)> positions = CookPositionsPosPos(buffer, posCount);
             foreach (BenchmarkIteration iteration in Benchmark.Iterations)
             {
                 int localInt = 0;
                 using (iteration.StartMeasurement())
                 {
                     for (int i = 0; i < Benchmark.InnerIterationCount; i++)
                     {
-                        ReadOnlySequence<byte> temp = buffer.Slice(offset, sliceLen);
-                        localInt ^= temp.Start.GetInteger();
+                        foreach ((SequencePosition start, SequencePosition end) in positions)
```

Comment:
How much overhead does the enumeration add? If the cost of just enumerating the sequence positions is order of magnitude larger than slicing, then the slicing time would get dwarfed by it.

Can you please compare the run-time of an empty foreach loop and one with the slice and share the difference (of even just one test?

Reply:
I think we should test Slicing with different params. Anyway we use one and the same loop in all tests so cost of just enumerating the sequence is constant time. I can replaced it with loop over list for best performance.